src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -2282,6 +2282,8 @@ void PhaseIdealLoop::do_unroll(IdealLoopTree *loop, Node_List &old_new, bool adj
     // can edit it's inputs directly.  Hammer in the new limit for the
     // minimum-trip guard.
     assert(opaq->outcnt() == 1, "");
+    // Notify limit -> opaq -> CmpI, it may constant fold.
+    _igvn.add_users_to_worklist(opaq->in(1));
```

Comment:
Should we call this on `new_limit` instead once we replaced the input of `opaq` with `new_limit`?